### DISCUSSION TITLE
Expose an API to allow other extensions to spell check documents

### DISCUSSION THREAD

**Discussion by martin-braun** [streetsidesoftware/vscode-spell-checker#708] [New Feature Requests]
Let's imagine one is building a nice VSCode extension that provides a nice WYSIWYG editor for a specific file type, is it possible to provide support for this extension, so it works in such WYSIWYG editor by using an API that is provided by Code Spell Checker?

The idea is to enable spell checking without the need to maintain a secondary custom dictionary. Given the fact Code Spell Checker is the leading spell checker extension for VSCode, I couldn't welcome this enough.

**Comment by Jason3S** [comment]
An API endpoint that can spell check a document is very possible depending upon the requirements.

The simplest form would be to tell the spell checker to check a file and have it give you back the results.

It is much easier if it is in a format that the underlying spell checker ([cspell](https://github.com/streetsidesoftware/cspell)) understands. Currently supported dictionaries: [cspell-dicts](https://github.com/streetsidesoftware/cspell-dicts/tree/master/dictionaries)

The extension already has an API to allow other extensions add configuration settings.

In any case, adding something like this desirable for supporting Jupyter Notebooks.

**Reply by martin-braun** [reply]
Thank you for your response. Right now I'm asking for an author of another extension in the hope that it will cause motivation for such an implementation. I made an issue relating this here: https://github.com/TobiasTao/vscode-md/issues/27

This is a great WYSIWYG editor for markdown files, so not dealing with a binary format. I think it would be great to have at least an API to send full documents and receive information where spell errors are located. Adding custom words to the dictionary itself should be easy, by modifying the settings of Code Spell Checker in the settings.json, so no endpoint needed on that.

> The simplest form would be to tell the spell checker to check a file and have it give you back the results.

Is this already possible or can I convert this discussion to an issue? Hopefully @TobiasTao will pick up on that, but this would bring value to Code Spell Checker nonetheless.
